src/plots/cartesian/autorange.js

```diff
@@ -174,15 +189,68 @@ function makePadFn(ax) {
     return function getPad(pt) { return pt.pad + (pt.extrapad ? extrappad : 0); };
 }
 
-function doAutoRange(ax) {
+function concatExtremes(gd, ax, ext) {
+    var i;
+    var out = [];
+
+    var fullData = gd._fullData;
+
+    // should be general enough for 3d, polar etc.
+
+    for(i = 0; i < fullData.length; i++) {
+        var trace = fullData[i];
+        var extremes = trace._extremes;
+
+        if(trace.visible === true) {
+            if(Registry.traceIs(trace, 'cartesian')) {
+                var axId = ax._id;
+                if(extremes[axId]) {
+                    out = out.concat(extremes[axId][ext]);
```

Comment:
I think my https://github.com/plotly/plotly.js/pull/2849#discussion_r205597761 and https://github.com/plotly/plotly.js/pull/2849#discussion_r205625136 still hold, and we should prune these the same way as in `findExtremes` to minimize `O(min.length * max.length)` in `doAutoRange`.

Reply:
oh sorry, you did that in https://github.com/plotly/plotly.js/pull/2860/commits/a0bfaf36fe7406b3868ed882c09c73e71074999f 🎉  and also 🌴  🏆 💯

Reply:
Yeah I tried to squash 72f06a6 into  a0bfaf3, but that was too much of a :face_with_head_bandage: